### DISCUSSION TITLE
How to count the number of runing tasks

### DISCUSSION THREAD

**Discussion by baoyachi** [tokio-rs/tokio#5181] [Q&A]
How to know how many tasks are running on the runtime of tokio

Can tokio provide functions similar to golang pprof

* https://github.com/golang/go/blob/master/src/runtime/pprof/pprof.go

**Comment by Darksonn** [comment]
Please see the [`RuntimeMetrics`][1] type and the [`tokio-metrics`][2] crate.

[1]: https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html
[2]: https://docs.rs/tokio-metrics/

**Reply by baoyachi** [reply]
Hi, @Darksonn  
I found out that [tokio-metrics](https://github.com/tokio-rs/tokio-metrics)  project is in a state of interruption, will the tokio team continue to develop and maintain it? because it has not been updated for 3 months.

**Reply by Darksonn** [reply]
The `tokio-metrics` crate has not been abandoned.

**Reply by Noah-Kennedy** [reply]
The reason that it hasn't seen any updates in 3 months is because there hasn't been much in the way of proposed changes or additions.
